packages/next-swc/crates/core/src/styled_jsx/mod.rs

```diff
@@ -652,34 +678,48 @@ fn get_style_expr(el: &JSXElement) -> &Expr {
                 .struct_span_err(
                     el.span,
                     &format!(
-                        "Expected one child under JSX style tag, but got {} (eg: <style \
-                         jsx>{{`hi`}}</style>)",
+                        "Expected one child under JSX style tag, but got {}.\nRead more: https://nextjs.org/docs/messages/invalid-styled-jsx-children",
                         non_whitespace_children.len()
                     ),
                 )
                 .emit()
         });
-        panic!("styled-jsx style error");
+        bail!("styled-jsx style error");
     }
 
     if let JSXElementChild::JSXExprContainer(JSXExprContainer {
         expr: JSXExpr::Expr(expr),
         ..
     }) = non_whitespace_children[0]
     {
-        return &**expr;
+        let expr_value = &**expr;
```

Comment:
This variable declaration seems redundant.
Is there a reason to use a variable instead of matching `&**expr` directly?